modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -2461,6 +2467,21 @@ private CacheGroupContext startCacheGroup(
         boolean persistenceEnabled = recoveryMode || sharedCtx.localNode().isClient() ? desc.persistenceEnabled() :
             dataRegion != null && dataRegion.config().isPersistenceEnabled();
 
+        EntryCompressionStrategy compressionStrategy = null;
+
+        Map<Class<? extends EntryCompressionConfiguration>,
```

Comment:
It is really unreadable snippet :) What if we just replace usage below like this:

```java
 if (compressionConfiguration != null) {
            compressionStrategy = CU.entryCompressionStrategies(ctx).get(compressionConfiguration.getClass())
                .apply(compressionConfiguration);

            compressionStrategy.start(ctx, cfg);
 }
```